Sources/Bugsnag/Bugsnag.swift

```diff
@@ -85,14 +86,30 @@ struct BugsnagRequest: Encodable {
     let referer: String
     let url: String
 
-    init(httpRequest: HTTPRequest) {
-        self.body = httpRequest.body.data.flatMap { String(data: $0, encoding: .utf8) }
+    init(httpRequest: HTTPRequest, keyFilters: [String]) {
+        self.body = BugsnagRequest.filter(httpRequest.body, using: keyFilters)
         self.clientIp = httpRequest.remotePeer.hostname
         self.headers = Dictionary(httpRequest.headers.map { $0 }) { first, second in second }
         self.httpMethod = httpRequest.method.string
         self.referer = httpRequest.remotePeer.description
         self.url = httpRequest.urlString
     }
+
+    static private func filter(_ body: HTTPBody, using filters: [String]) -> String? {
+        guard
+            let data = body.data,
+            let unwrap = try? JSONSerialization.jsonObject(
+                with: data, options: []
+            ) as? [String: Any],
+            let jsonObject = unwrap
+        else {
+            return body.data.flatMap { String(data: $0, encoding: .utf8) }
+        }
+
+        let filtered = jsonObject.filter { !filters.contains($0.key) }
```

Comment:
possibly faster would be `jsonObject.removeAll { filters.contains($0.key) }` which would require `var jsonObject = unwrap` above.